mods/cnc/maps/shellmap/shellmap.lua

```diff
@@ -1,3 +1,362 @@
+	end)
+
+	Trigger.OnIdle(commandos[2], function()
+			if(Map.NamedActor("Turret2") ~= nil) then
+				commandos[2].Demolish(Turret2)
+			end
+			if(Map.NamedActor("beachdude1") ~= nil or Map.NamedActor("beachdude2") ~= nil) then
+				commandos[2].AttackMove(CommMove.Location)
+			end
+			if(Map.NamedActor("Turret4") ~= nil) then
+				commandos[2].Demolish(Turret4)
+				commandos[2].Move(CommFin.Location)
+			end
+			commandos[2].Move(CommFin.Location)
+	end)
+
+	Trigger.OnKilled(Turret1, function()
+			beachdude1.AttackMove(BeachDudeMove.Location)
+			beachdude2.AttackMove(BeachDudeMove.Location)
+	end)
+
+	Trigger.OnKilled(Turret3, function()
+			commandos[1].Stop()
+			commandos[1].ScriptedMove(CommFin.Location)
+			Trigger.Clear(commandos[1], "OnIdle")
+			Trigger.AfterDelay(25, Beachhead)
+	end)
+
+	Trigger.OnKilled(Turret4, function()
+			commandos[2].Stop()
+			commandos[2].ScriptedMove(CommFin.Location)
+			Trigger.Clear(commandos[2], "OnIdle")
+	end)
+	Trigger.OnAllKilled({beachdude1, beachdude2}, function()
+		local trans = Reinforcements.ReinforceWithTransport(gdi, "tran", nil, {EvacEntry.Location, EvacLand.Location})[1]
+		Trigger.OnIdle(trans, function()
+			if(trans.HasPassengers) then
+				trans.Wait(25)
+				trans.CallFunc(function() PlayMusic() end)
+				trans.Move(EvacEntry.Location)
+				trans.Destroy()
+			else
+				commandos[1].ScriptedMove(FlareHere.Location)
+				commandos[2].ScriptedMove(FlareHere.Location)
+				commandos[1].EnterTransport(trans)
+				commandos[2].EnterTransport(trans)
+			end
+		end)
+	end)
+end
+
+Beachhead = function()
+	--todo: Make custom reinforce function to handle LST Dropoff properly
+	local ltank = Reinforcements.ReinforceWithTransport(gdi, "lst", {"htnk"}, LSTLEntryPath, LSTLExitPath, UnloadLST)[2]
+	Trigger.OnAddedToWorld(ltank[1], function(unit) LeftTankAction(unit) end)
+	Trigger.AfterDelay(55, function()
+		local mcv = Reinforcements.ReinforceWithTransport(gdi, "lst", {"mcv"}, LSTMEntryPath, LSTMExitPath, UnloadLST)[2]
+		Trigger.OnAddedToWorld(mcv[1], function(unit) MCVAction(unit) end)
+	end)
+	Trigger.AfterDelay(110, function()
+		local rtank = Reinforcements.ReinforceWithTransport(gdi, "lst", {"htnk"}, LSTREntryPath, LSTRExitPath, UnloadLST)[2]
+		Trigger.OnAddedToWorld(rtank[1], function(unit) RightTankAction(unit) end)
+	end)
+end
+
+RightTankAction = function(tank)
+	tank.ScriptedMove(LSTRLand.Location - CVec.New(0,4))
+	tank.ScriptedMove(LSTLLand.Location - CVec.New(12,4))
+	MakeUnitInvincible(tank)
+end
+
+LeftTankAction = function(tank)
+	tank.ScriptedMove(LSTLLand.Location - CVec.New(0,5))
+	tank.ScriptedMove(LSTLLand.Location - CVec.New(11,5))
+	MakeUnitInvincible(tank)
+end
+
+MCVAction = function(mcv)
+	--mcv.ScriptedMove(LSTMLand.Location - CVec.New(0,1))
+	mcv.Wait(250)
+	mcv.ScriptedMove(MCVDeploy.Location)
+	mcv.CallFunc(function() Trigger.AfterDelay(100, function() BuildBase() end) end)
+	Trigger.OnIdle(mcv, function() mcv.Deploy()	end)
```

Comment:
This trigger is unnecessary as well. Just call `mcv.Deploy()`, the action will be queued.